tests/fuzz/mesh_fuzzer.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint: golint
+package fuzz
+
+import (
+	"istio.io/istio/pkg/config/mesh"
+)
+
+func FuzzParseMeshNetworks(data []byte) int {
+	mn, err := mesh.ParseMeshNetworks(string(data))
+	if err != nil {
+		return 0
+	}
+	mesh.ResolveHostsInNetworksConfig(mn)
```

Comment:
This is doing a real network DNS lookup. Is that legal/a good idea in a fuzzer?

Reply:
Thanks for the heads up. We don't want to do that. I have removed that part.